File: src/main/java/com/danidipp/sneakypocketbase/AsyncPocketbaseEvent.java
package com.danidipp.sneakypocketbase;

import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;

/** Realtime PocketBase event whose interface contains only JDK and Bukkit types. */
public final class AsyncPocketbaseEvent extends Event {
    public enum Action {
        CONNECT,
        CREATE,
        UPDATE,
        DELETE
    }

    private static final HandlerList HANDLERS = new HandlerList();

    private final Action action;
    private final String collectionName;
    private final String recordJson;

    public AsyncPocketbaseEvent(
        final boolean async,
        final Action action,
        final String collectionName,
        final String recordJson
    ) {
        super(async);
        this.action = action;
        this.collectionName = collectionName;
        this.recordJson = recordJson;
    }

    public Action getAction() {
        return this.action;
    }

    public String getCollectionName() {
        return this.collectionName;
    }

    public String getRecordJson() {
        return this.recordJson;
    }

    @Override
    public HandlerList getHandlers() {
        return HANDLERS;
    }

    public static HandlerList getHandlerList() {
        return HANDLERS;
    }
}

File: src/main/kotlin/com/danidipp/sneakypocketbase/AsyncPocketbaseEvent.kt
package com.danidipp.sneakypocketbase

import io.github.agrevster.pocketbaseKotlin.services.RealtimeService
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class AsyncPocketbaseEvent(
    async: Boolean,
    var action: RealtimeService.RealtimeActionType,
    var collectionName: String,
    var data: RealtimeService.MessageData
): Event(async) {
    override fun getHandlers(): HandlerList {
        return HANDLERS
    }

    companion object {
        private val HANDLERS = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList {
            return HANDLERS
        }
    }
}